ddsp/training/docker/ddsp_ai_platform.py

```diff
@@ -0,0 +1,270 @@
+                    'training. Can be different than the save_dir.')
+flags.DEFINE_string('config_path', '',
+                    'Configuration file for AI Platform training')
+flags.DEFINE_string('project_id', '', 'GCP Project ID')
+flags.DEFINE_string('region', '', 'GCP region for running the training')
+flags.DEFINE_string('batch_size', '', 'Batch size')
+flags.DEFINE_string('learning_rate', '', 'Learning rate')
+flags.DEFINE_string('num_steps', '', 'Number of training steps')
+flags.DEFINE_string('steps_per_summary', '', 'Steps per summary')
+flags.DEFINE_string('steps_per_save', '', 'Steps per save')
+flags.DEFINE_string('early_stop_loss_value', '',
+                   'Early stopping. When the total_loss reaches below this '
+                   'value training stops.')
+
+def prompt_gs_path(message, required=False):
+  if required:
+    message += ' (Required): '
+  else:
+    message += ': '
+  path = input(message)
+  if required or path:
+    path = check_bucket(path, message)
+  return path
+
+def prompt_local_path(message):
+  path = input(message)
+  if path:
+    path = check_local_file(path, message)
+  else:
+    path = './config_multiple_vms.yaml'
+  return path
+
+def prompt_value(message, arg):
+  message += ' (Default: ' + arg + '): '
+  value = input(message)
+  if not value:
+    value = arg
+  return value
+
+def check_local_file(file_path, message):
+  while not os.path.isfile(file_path):
+    print('File doesn\'t exist.')
+    file_path = input(message)
+  return file_path
+
+def check_bucket(bucket_name, message):
+  while not bucket_name.startswith('gs://'):
+    print(f'Path {bucket_name} is not a valid GCS bucket. '
+          'Bucket paths must start with gs://.')
+    bucket_name = input(message)
+  while True:
+    try:
+      name = bucket_name.split('gs://')[1]
+      if '/' in name:
+        name = name.split('/')[0]
+      storage.Client().get_bucket(name)
+      break
+    except:
+      print('Bucket doesn\'t exist or can\'t be accessed.')
+      bucket_name = input(message)
+  return bucket_name
+
+def check_project_id(project_id):
+  check_command = 'gcloud projects describe ' + project_id
+  while not subprocess.getoutput(check_command).startswith('createTime'):
+    print('Project ID not valid.')
+    project_id = input('Project ID: ')
+    check_command = 'gcloud projects describe ' + project_id
+  return project_id
+
+def get_project_id():
+  project_id = subprocess.getoutput('gcloud config get-value project')
+  if project_id == '(unset)':
+    project_id = input('Project ID: ')
+    project_id = check_project_id(project_id)
+  return project_id
+
+def get_region():
+  region = subprocess.getoutput('gcloud config get-value compute/region')
+  if region == '(unset)':
+    region = prompt_value('Region for training the model', 'europe-west4')
+  return region
+
+def get_input():
+  """Gathers input from user"""
+  if FLAGS.data_path:
+    FLAGS.data_path = check_bucket(FLAGS.data_path,
```

Comment:
FLAGS are a global so should just be set by the absl.flags. I think the better thing would be to define the local variable, ex:

```
msg = 'Path to training dataset directory'
if FLAGS.data_path:
  data_path = check_bucket(FLAGS.data_path, msg)
else:
  data_path = propt_gs_path(msg, required=True)
```